addons/account/models/account_move.py

```diff
@@ -675,7 +675,8 @@ def _compute_name(self):
         highest_name = self[0]._get_last_sequence(lock=False) if self else False
 
         for move in self:
-            if not highest_name and move == self[0] and not move.posted_before and move.date and (not move.name or move.name == '/'):
+            if not highest_name and move == self[0] and not move.posted_before and move.date and \
+                    ((not move.name or move.name == '/') or move.move_type == 'entry'):
```

Comment:
Why should it be different on an invoice as on a misc entry?  I kind of think we just need something similar as in L686 - L692 for journals.  Although not even: ideally we would have an onchange on the journal that resets the name and calls this compute. @guva-odoo

Reply:
@william-andre

Reply:
> Why should it be different on an invoice as on a misc entry?

It shouldn't. It is probably a field protection issue again; see https://github.com/odoo/odoo/pull/116829

Reply:
> I kind of think we just need something similar as in L686 - L692 for journals.

It is precisely because we enter in the condition L686-692 that the sequence doesn't change.

Before e31c5767c894a92d22fa8201bc1ccf34b052a336, we entered in the first condition, this is why I think something should be made at this place